internal/document/frontmatter.go

```diff
@@ -87,19 +87,21 @@ func newFrontmatter() *Frontmatter {
 	}
 }
 
-// Marshal returns a marshaled frontmatter including triple-dashed lines.
-// If the identity is required, but Frontmatter is nil, a new one is created.
-func (f *Frontmatter) Marshal(requireIdentity bool) ([]byte, error) {
+// Marshal returns a marshaled frontmatter including triple-dashed lines and a document ID.
+// If the identity is required, but Frontmatter is nil, a new one including ID is created.
+func (f *Frontmatter) Marshal(requireIdentity bool) ([]byte, string, error) {
```

Comment:
Remark: In the ideal world we would strive for following signature: `Marshal() ([]byte, error)`.

Reply:
I know, but it would require me to rethink/rewrite a large part of the prepopulating frontmatter scenario which makes other things super complicated.

Reply:
Still need to do more integration testing with the extension before committing to this change, fwiw.